content/blog/0020-schema-release-0.67.mdx

```diff
@@ -0,0 +1,551 @@
+console.log(PersonSchema.make({ name: "John" }))
+// => { age: 0, name: 'John' }
+```
+
+Defaults are **lazily evaluated**, meaning that a new instance of the default is generated every time the constructor is called:
+
+```ts
+import { Schema } from "@effect/schema"
+
+const PersonSchema = Schema.Struct({
+  name: Schema.NonEmpty,
+  age: Schema.Number.pipe(
+    Schema.propertySignature,
+    Schema.withConstructorDefault(() => 0)
+  ),
+  timestamp: Schema.Number.pipe(
+    Schema.propertySignature,
+    Schema.withConstructorDefault(() => new Date().getTime())
+  )
+})
+
+console.log(PersonSchema.make({ name: "name1" }))
+// => { age: 0, timestamp: 1714232909221, name: 'name1' }
+console.log(PersonSchema.make({ name: "name2" }))
+// => { age: 0, timestamp: 1714232909227, name: 'name2' }
+```
+
+Note how the `timestamp` field varies.
+
+Defaults can also be applied using the `Class` API:
+
+```ts
+import { Schema } from "@effect/schema"
+
+class Person extends Schema.Class<Person>("Person")({
+  name: Schema.NonEmpty,
+  age: Schema.Number.pipe(
+    Schema.propertySignature,
+    Schema.withConstructorDefault(() => 0)
+  ),
+  timestamp: Schema.Number.pipe(
+    Schema.propertySignature,
+    Schema.withConstructorDefault(() => new Date().getTime())
+  )
+}) {}
+
+console.log(new Person({ name: "name1" }))
+// => Person { age: 0, timestamp: 1714400867208, name: 'name1' }
+console.log(new Person({ name: "name2" }))
+// => Person { age: 0, timestamp: 1714400867215, name: 'name2' }
+```
+
+## Default Decoding Values
+
+Our new `Schema.withDecodingDefault` combinator makes it easy to handle the optionality of a field during the **decoding process**.
+
+```typescript
+import { Schema } from "@effect/schema"
+
+const schema = Schema.Struct({
+  a: Schema.optional(Schema.String).pipe(Schema.withDecodingDefault(() => ""))
+})
+
+console.log(Schema.decodeUnknownSync(schema)({}))
+// => { a: '' }
+console.log(Schema.decodeUnknownSync(schema)({ a: undefined }))
+// => { a: '' }
+console.log(Schema.decodeUnknownSync(schema)({ a: "a" }))
+// => { a: 'a' }
+```
+
+If you want to set default values both for the decoding phase and for the default constructor, you can use `Schema.withDefaults`:
+
+```ts
+import { Schema } from "@effect/schema"
+
+const schema = Schema.Struct({
+  a: Schema.optional(Schema.String).pipe(
+    Schema.withDefaults({
+      decoding: () => "",
+      constructor: () => "-"
+    })
+  )
+})
+
+console.log(Schema.decodeUnknownSync(schema)({}))
+// => { a: '' }
+console.log(Schema.decodeUnknownSync(schema)({ a: undefined }))
+// => { a: '' }
+console.log(Schema.decodeUnknownSync(schema)({ a: "a" }))
+// => { a: 'a' }
+
+console.log(schema.make({})) // => { a: '-' }
+console.log(schema.make({ a: "a" })) // => { a: 'a' }
+```
+
+## Introducing Schemas as Classes
+
+We've introduced a new method for **defining schemas using classes**.
+This approach provides **opaque schema types**, offering a clearer representation.
```

Comment:
some feedback about naming and some limitations https://discord.com/channels/795981131316985866/1231120836195385355/1237109976426872923